### PR TITLE
Centralize disallow inner text handling in ParseHalper

### PR DESCRIPTION
**Pull request by robmen**
Fixes wixtoolset/issues#6237

### CODE REVIEW

File: src/WixToolset.Core/ExtensibilityServices/ParseHelper.cs [hunk 1]
```diff
@@ -714,17 +729,31 @@ public IComponentKeyPath ParsePossibleKeyPathExtensionElement(IEnumerable<ICompi
...
+                    }
+                    else
+                    {
+                        this.ParseExtensionElement(extensions, intermediate, section, element, childElement);
+                    }
                 }
                 else
                 {
-                    this.ParseExtensionElement(extensions, intermediate, section, element, child);
+                    checkInnerText = true;
```
Comment (rseanhall) [inline src/WixToolset.Core/ExtensibilityServices/ParseHelper.cs:749]: Is this an opinion we want to unconditionally push on all extensions that want to call `ParseForExtensionElements`?

File: src/WixToolset.Core/Bind/TransferFilesCommand.cs [hunk 2]
```diff
@@ -157,19 +156,13 @@ public void Execute()
...
-                    {
-                        new FileInfo(file).SetAccessControl(aclReset);
-                    }
+                    FileSystem.ResetAcls(destinationFiles);
                 }
-                catch
+                catch (Exception e)
                 {
-                    this.Messaging.Write(WarningMessages.UnableToResetAcls());
+                    this.Messaging.Write(WarningMessages.UnableToResetAcls(e.Message));
```
Comment (rseanhall) [inline src/WixToolset.Core/Bind/TransferFilesCommand.cs:165]: I hope the interesting information isn't in inner exceptions.

File: src/WixToolset.Core/ExtensibilityServices/ParseHelper.cs (hunk 1 above)
Comment (robmen) [reply src/WixToolset.Core/ExtensibilityServices/ParseHelper.cs:749]: Yes. Absolutely.

File: src/WixToolset.Core/Bind/TransferFilesCommand.cs (hunk 2 above)
Comment (robmen) [reply src/WixToolset.Core/Bind/TransferFilesCommand.cs:165]: Me too. So far, so good.

**Pull request merged**